imports/plugins/core/orders/server/methods/capturePayments.js

```diff
@@ -35,16 +38,14 @@ export default function capturePayments(orderId) {
 
   // find the payment based on shopId
   const { _id: groupId, payment } = order.shipping.find((group) => group.shopId === shopId);
-  const { mode, processor, status, transactionId } = payment;
-
-  if (mode === "capture" && status === "approved" && processor) {
-    // Grab the amount from the shipment, otherwise use the original amount
-    const processorLowercase = processor.toLowerCase();
+  const { mode, name, status, transactionId } = payment;
 
+  if (mode === "capture" && status === "approved" && name) {
     let result;
     let error;
     try {
-      result = Meteor.call(`${processorLowercase}/payment/capture`, payment);
+      const context = Promise.await(getGraphQLContextInMeteorMethod(Reaction.getUserId()));
+      result = Promise.await(paymentMethods[name].functions.capturePayment(context, payment));
```

Comment:
@ajporlante This is correct, but it's likely that at some point people could switch payment providers and end up removing the old one, in which case `paymentMethods[name]` will be undefined and you'll get a cryptic error. I'd rather give them a nice error.

You could add a function in `pluginRegistration.js`:

```js
export function getPaymentMethodConfigByName(name) {
  const config = paymentMethods[name];
  if (!config) {
    throw new Error(`Configuration not found for ${name} payment method. Did you remove the plugin that provides this payment method?`);
  }
  return config;
}
```

And then import and use `getPaymentMethodConfigByName(name)` in these methods.